### PR TITLE
[Merged by Bors] - chore(MeasureTheory): golf

### PR DESCRIPTION
**Pull request by urkud**
Also, these `simp` calls are better compatible
with a future refactor using `FunLike` for measures.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by grunweg — APPROVED** [review]

File: Mathlib/MeasureTheory/Measure/MeasureSpace.lean [hunk 1]
```diff
@@ -1627,14 +1614,12 @@ protected theorem smul [Monoid R] [DistribMulAction R ℝ≥0∞] [IsScalarTower
   simp only [h hνs, smul_eq_mul, smul_apply, smul_zero]
 #align measure_theory.measure.absolutely_continuous.smul MeasureTheory.Measure.AbsolutelyContinuous.smul
 
-protected lemma add (h1 : μ₁ ≪ ν) (h2 : μ₂ ≪ ν') : μ₁ + μ₂ ≪ ν + ν' := by
-  intro s hs
-  simp only [add_toOuterMeasure, OuterMeasure.coe_add, Pi.add_apply, add_eq_zero] at hs ⊢
+protected lemma add (h1 : μ₁ ≪ ν) (h2 : μ₂ ≪ ν') : μ₁ + μ₂ ≪ ν + ν' := fun s hs ↦ by
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Measure/MeasureSpace.lean:1617]: I think I prefer the spelling with `intro s hs` than the spelling with `fun`, as it doesn't mix term mode and tactic mode. Same thing below.

**Review by sgouezel — COMMENTED** [review]
The `simp only` calls were certainly generated by `simp?`. Is it possible to adjust some simp prorities to make sure that `simp?` calls correspond to the shorter versions you're using in the PR?

File: Mathlib/MeasureTheory/Measure/MeasureSpace.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/MeasureTheory/Measure/MeasureSpace.lean:1617]:
```suggestion
protected lemma add (h1 : μ₁ ≪ ν) (h2 : μ₂ ≪ ν') : μ₁ + μ₂ ≪ ν + ν' := by
  intro s hs
```

File: Mathlib/MeasureTheory/Measure/MeasureSpace.lean [hunk 2]
```diff
@@ -1627,14 +1614,12 @@ protected theorem smul [Monoid R] [DistribMulAction R ℝ≥0∞] [IsScalarTower
...
+  simp only [coe_add, Pi.add_apply, add_eq_zero] at hs ⊢
   exact ⟨h1 hs.1, h2 hs.2⟩
 
-lemma add_right (h1 : μ ≪ ν) (ν' : Measure α) : μ ≪ ν + ν' := by
-  intro s hs
-  simp only [add_toOuterMeasure, OuterMeasure.coe_add, Pi.add_apply, add_eq_zero] at hs ⊢
+lemma add_right (h1 : μ ≪ ν) (ν' : Measure α) : μ ≪ ν + ν' := fun s hs ↦ by
```
Comment (eric-wieser) [inline Mathlib/MeasureTheory/Measure/MeasureSpace.lean:1621]:
```suggestion
lemma add_right (h1 : μ ≪ ν) (ν' : Measure α) : μ ≪ ν + ν' := by
  intro s hs
```

**Comment by eric-wieser** [comment]
> The `simp only` calls were certainly generated by `simp?`. Is it possible to adjust some simp prorities to make sure that `simp?` calls correspond to the shorter versions you're using in the PR?

Any action needed here, or should we just merge this as is?

**Comment by urkud** [comment]
I'll make some lemmas `@[simp high]` later today and see if `simp?` uses lemmas I want it to use. Sorry for the delay and thanks @eric-wieser for your help!

**Comment by urkud** [comment]
@sgouezel I migrated `Measure` and `OuterMeasure` to `FunLike` and reverted some worthless micro golfs.

**Pull request closed without merging**